File: codegames/src/main/java/lib/codegames/graphics/SizeCG.java
package lib.codegames.graphics;

import android.content.res.Resources;
import android.util.DisplayMetrics;

public final class SizeCG {

    private static final DisplayMetrics displayMetrics = Resources.getSystem().getDisplayMetrics();
    private static final float density = displayMetrics.density;

    public static int dp2px(int dp) {
        return (int) (dp * density);
    }

    public static int px2dp(int px) {
        return (int) (px / density);
    }

    public static int getScreenWidthPX() {
        return displayMetrics.widthPixels;
    }

    public static int getScreenHeightPX() {
        return displayMetrics.heightPixels;
    }

}

File: codegames/src/main/java/lib/codegames/graphics/SizeCG.kt
package lib.codegames.graphics

import android.content.res.Resources
import android.util.DisplayMetrics

class SizeCG(val width: Int, val height: Int) {

    companion object {

        private val displayMetrics = Resources.getSystem().displayMetrics
        private val density = displayMetrics.density

        val screenWidthPX: Int
            get() = displayMetrics.widthPixels

        val screenHeightPX: Int
            get() = displayMetrics.heightPixels

        fun dp2Px(dp: Int): Int {
            return (dp * density).toInt()
        }

        fun px2Dp(px: Int): Int {
            return (px / density).toInt()
        }

    }

}
